lib/foreman_maintain/package_manager/yum.rb

```diff
@@ -0,0 +1,129 @@
+module ForemanMaintain::PackageManager
+  class Yum < Base
+    VERSIONLOCK_START_CLAUSE = '## foreman-maintain - start'.freeze
+    VERSIONLOCK_END_CLAUSE = '## foreman-maintain - end'.freeze
+    VERSIONLOCK_CONFIG_FILE = '/etc/yum/pluginconf.d/versionlock.conf'.freeze
+    VERSIONLOCK_DEFAULT_LIST_FILE = '/etc/yum/pluginconf.d/versionlock.list'.freeze
+
+    def foreman_related_packages
+      installed = sys.execute("repoquery -a --qf='%{envra} %{repo.baseurl}'").
+                  split("\n"). map(&:split)
+      repos_to_lock = []
+      foreman_repo = installed.find { |package| package[0] =~ /foreman-installer-\d+/ }[1]
+      repos_to_lock << foreman_repo
+      installed.inject([]) do |related, package|
```

Comment:
I also feel that we should also lock rhscl, ansible and puppet repositories because some of Satellite bits should be depending on these repositories, but I am not very sure how frequently we update these three repositories, if those are not having frequent upgrade and just locking packages from Satellite repository does not break things then I think there is no need.

Reply:
For a record: after some discussion we decided not to lock collections and other repos for now.